README
######################################################################
## File: $Id: README,v 1.1 2002/09/09 01:34:10 spadkins Exp $
######################################################################

WHAT IS THIS?

This is the App-Context distribution.
For more information, see the web pages at

   http://www.officevision.com/pub/App-Context

and/or join the mailing list at

   p5ee@perl.org

The mailing list is described at

   http://p5ee.perl.org/mailinglist/
   http://lists.perl.org/showlist.cgi?name=p5ee

HOW DO I INSTALL IT?

To install this module, cd to the directory that contains this README
file and type the following:

   perl Makefile.PL
   make
   make test
   make install



CHANGES
#########################################
# CHANGE LOG
#########################################

VERSION 0.967
 x App::Request::CGI     - added the url() method to support putting the URL into the form tag in App::Widget
 x App::Service          - added get_sym_label() method for various uses in turning a symbol into a label
 x App::Exceptions       - removed call to do_trace on Exception::Class::Base (the method has been removed)
 x App::Context          - Fixed bug with a defined variable
 x App::Context          - add authentication_class option from app.conf
 x App::Context::POE::Server - added a storable option (poe_storable) to allow for communicating more than just a string value back and forth between parent and child processes
 x App::ValueDomain      - Added sub get_label and _load_label to allow for loading of entries in a ValueDomain one at a time
 x App::Storable         - protect against varying versions of Storable which may cause deserialization exceptions
 x App::Serializer::Json - read and write JSON
 x App::Serializer::Html - serialize output as HTML for debugging/viewing
 x App::Context::HTTP    - add support for RPC and REST (in addition to web app support)
 x App::Context::service_exists() - detect if a service exists in the config/session (generalizes session_object_exists())

VERSION 0.9661
 x updated the dependencies so that the CPAN tests don't fail

VERSION 0.966
 x add SharedDatastore as a useful service (with a Repository-based implementation)
 x improved support for "temporary" services (named "temporary" or with the {temporary} arg)
   (a "temporary service" is akin to a stateless session bean in Java)
 x add support for including/overlaying additional config files based on values present in
   the %$options hash or when a particular named service is instantiated
 x App::Context::POE::Server
 x App::Context::POE::ClusterController, App::Context::POE::ClusterNode

VERSION 0.965
 x add UI timing log, activated by "app.Context.timer" option

VERSION 0.964
 x mostly documentation updates
 x moved Apache::Framework::App to Apache::App
 x trying to improve Makefile.PL to install things in the right place (app in cgi-bin)

VERSION 0.963
 x App->use() now detects entries in the symbol table to inhibit an attempt to load the module
 x enhance App::Service::substitute() to understand default syntax (i.e. {username:NULL})

VERSION 0.962
   I don't remember what 0.962 was all about (minor)

VERSION 0.961
 x add per-module (subversion) $VERSION numbers

VERSION 0.96
 x added &App::in_debug_scope and &App::debug_indent()
 x context->widget() now defaults to class App::Widget rather than class App::SessionObject
 x trace enhancements
   * print 'undef' instead of '' for undefined values in traces
   * made trace behave better with newlines. added trace_justify.
   * fixed a trace leak (return without sub_exit())
   * trace and debug scopes can now be exclusive (i.e. --trace=!App::Context,App::Context::Cmd)
 x log enhancements
   * added log levels (and optional args to log())
 x {temporary} widgets
 x add permissions to widget display from URL and method calls, also {open_widget_urls} option
 x add App::Service::substitute()
 x default dates can be {today}+6

VERSION 0.95
 x Widget containership is defined in the widget name by dashes ("-") instead of dots (".") (better for CSS)
 x Removed dependency on App::Repository in t/ServiceRemote.t

VERSION 0.93
 x Initial release (first one I started keeping this change log, anyway)
 x Implemented App::Reference->overlay() more completely



Makefile.PL

######################################################################
## File: $Id: Makefile.PL 13307 2009-09-11 13:59:44Z spadkins $
######################################################################

use ExtUtils::MakeMaker;

my @programs = (
    "bin/app",
    #"bin/mason",
    "bin/call",
    "bin/app-apache",
    #"bin/app-netserver",
);

%opts = (
    "NAME"        => "App-Context",
    "DISTNAME"    => "App-Context",
    "VERSION"     => "0.967",
    "EXE_FILES"   => [ @programs ],
    "PREREQ_PM"   => {
        # "Apache"                             => "0.01",  # used for mod_perl integration
        "App::Options"                       => "0.01",  # for loading a startup configuration file
        "CGI"                                => "0.01",  # tbd
        "CGI::Carp"                          => "0.01",  # tbd
        "Class::Data::Inheritable"           => "0.01",  # [prereq for Exception::Class]
        "Compress::Zlib"                     => "0.01",  # for compressed serialization and browser responses
        "Data::Dumper"                       => "0.01",  # used for debugging
        "Date::Parse"                        => "0.01",  # date support
        "Date::Format"                       => "0.01",  # date support
        "Devel::StackTrace"                  => "0.01",  # [prereq for Exception::Class]
        "Digest::SHA"                        => "0.01",  # tbd
        "Exception::Class"                   => "0.01",  # allows exception hierarchies
        "HTTP::Status"                       => "0.01",  # tbd
        "IO::Handle"                         => "0.01",  # tbd
        "IO::Socket"                         => "0.01",  # tbd
        "IO::Socket::INET"                   => "0.01",  # tbd
        "MIME::Base64"                       => "2.1",   # used for turning binary (serialized?) data into text (Sessions)
        "Pod::Checker"                       => "0.01",  # tbd
        # "POE"                                => "0.01",  # tbd
        # "POE::Component::IKC"                => "0.01",  # tbd
        # "POE::Component::Server::SimpleHTTP" => "0.01",  # tbd
        "POSIX"                              => "0.01",  # tbd
        "Socket"                             => "0.01",  # tbd
        "Storable"                           => "0.01",  # used for serialization everywhere
        "Sys::Hostname"                      => "0.01",  # tbd
        "Test::More"                         => "0.01",  # tbd
        "Time::HiRes"                        => "0.01",  # tbd
        "XML::Simple"                        => "0.01",  # tbd
        "YAML"                               => "0.01",  # tbd
    },
);

######################################################################
# MAKE THE MAKEFILE
######################################################################

WriteMakefile(%opts);

sub MY::postamble {
    return <<EOF;

install ::
	@\$(RM_F) "\$(PREFIX)/cgi-bin/app"
	@\$(CP) "\$(PREFIX)/bin/app" "\$(PREFIX)/cgi-bin/app"

EOF
}

